### PR TITLE
add stack.md outline with modified CFU section

### PR DESCRIPTION
**Pull request by rudri182**
Created new branch with CFU section questions for stacks. The issue is #700

### CODE REVIEW

**Comment by rudri182** [comment]
I have made new PR to **check for understanding** section and it only contains changes in the CFU section.

File: data-structures/stack.md [hunk 1]
```diff
@@ -65,9 +65,20 @@ var Stack = function() {}
...
 Find a partner and show each other your own Stack class. Explain how the class you wrote works.
 
+Make a pair of two and ask the following questions to each-other.
+
+- What is difference between stack and simple array?
+- Which method does the stack follows?(LIFO)
+- Which methods are used in the stack data structure?
+- What is run time complexity of the stack?
+- How to make simple class as a stack class?
+- What is the meaning of '_' (*underscore*) in the beginning of the variable name? Refer [Playing with data structures in Javascript — Stack](https://blog.cloudboost.io/playing-with-data-structures-in-javascript-stack-a55ebe50f29d)
```
Comment (alodahl) [inline data-structures/stack.md]: 😎

File: data-structures/stack.md [hunk 2]
```diff
@@ -65,9 +65,20 @@ var Stack = function() {}
...
+- Give some day-to-day examples where stack is used.
```
Comment (alodahl) [inline data-structures/stack.md]: lines 76 and 77 seem like the same question to me.

File: data-structures/stack.md [hunk 3]
```diff
@@ -65,9 +65,20 @@ var Stack = function() {}
 
 Find a partner and show each other your own Stack class. Explain how the class you wrote works.
 
+Make a pair of two and ask the following questions to each-other.
+
+- What is difference between stack and simple array?
+- Which method does the stack follows?(LIFO)
+- Which methods are used in the stack data structure?
+- What is run time complexity of the stack?
```
Comment (alodahl) [inline data-structures/stack.md]: of *a* stack

File: data-structures/stack.md [hunk 4]
```diff
@@ -65,9 +65,20 @@ var Stack = function() {}
 
 Find a partner and show each other your own Stack class. Explain how the class you wrote works.
 
+Make a pair of two and ask the following questions to each-other.
+
+- What is difference between stack and simple array?
```
Comment (alodahl) [inline data-structures/stack.md]: *a* stack and *a* simple array

File: data-structures/stack.md [hunk 5]
```diff
@@ -65,9 +65,20 @@ var Stack = function() {}
 
 Find a partner and show each other your own Stack class. Explain how the class you wrote works.
 
+Make a pair of two and ask the following questions to each-other.
```
Comment (alodahl) [inline data-structures/stack.md]: Maybe replace with
"Next, ask each other the following questions:"

File: data-structures/stack.md (hunk 3 above)
Comment (alodahl) [inline data-structures/stack.md]: What is *the* runtime complexity of *a* stack?

File: data-structures/stack.md [hunk 6]
```diff
@@ -65,9 +65,20 @@ var Stack = function() {}
...
+
+- What is difference between stack and simple array?
+- Which method does the stack follows?(LIFO)
```
Comment (alodahl) [inline data-structures/stack.md]: could you use this if it has the same meaning?:
"Which method uses LIFO?"

File: data-structures/stack.md (hunk 5 above)
Comment (alodahl) [reply data-structures/stack.md]: We've already established that you're with a partner.

File: data-structures/stack.md (hunk 1 above)
Comment (rudri182) [reply data-structures/stack.md]: 😄

File: data-structures/stack.md (hunk 6 above)
Comment (rudri182) [reply data-structures/stack.md]: Yes, that's a nice suggestion. 🎉 I will make it.

**Comment by rudri182** [comment]
@alodahl your suggestions are great! Thank you for noticing such minor typos and grammar. I have changed it in the outline. ✏️  Please review it and suggest if there is more such type of additions.😃

**Pull request merged**